src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -2179,6 +2181,396 @@ impl Coordinator {
         };
     }
 
+    #[instrument]
+    pub(super) async fn sequence_explain_analyze(
+        &mut self,
+        mut ctx: ExecuteContext,
+        plan: plan::ExplainAnalyzePlan,
+        // TODO: What is this used for? Should we use it for anything? PLAN INSIGHTS seems maybe
+        // correct? But the others could be a no because we always want the introspection queries to
+        // use the auto route cluster?
+        target_cluster: TargetCluster,
+    ) {
+        // TODO: Not convinced this is correct. Can we use `ctx.session().pcx()`?
+        let now = self.now();
+        let catalog = self.owned_catalog();
+        let cluster_id = return_if_err!(
+            catalog.resolve_target_cluster(target_cluster, ctx.session()),
+            ctx
+        )
+        .id;
+        // TODO: Should this be passed to sequence_explain_analyze instead?
+        let otel_ctx = OpenTelemetryContext::obtain();
+        mz_ore::task::spawn(|| "sequence_explain_analyze", async move {
+            // EXPLAIN ANALYZE must be run as a single statement (see handle_execute where that is
+            // enforced). But we want all of the inner queries to be at the same timestamp with the
+            // correct timedomain, so explicitly start the session transaction.
+            return_if_err!(
+                ctx.session_mut().start_transaction(
+                    mz_ore::now::to_datetime(now),
+                    // Hard code these so that any session settings don't mess with them.
+                    Some(mz_sql::ast::TransactionAccessMode::ReadOnly),
+                    // TODO: Actually, should this be `None` so it picks up the current session var?
+                    // Need to think through what would happen to the various query results and if
+                    // that's what the user would want.
+                    Some(mz_sql::ast::TransactionIsolationLevel::StrictSerializable),
+                ),
+                ctx
+            );
+
+            let depends = match &plan.explainee {
+                Explainee::View(id)
+                | Explainee::MaterializedView(id)
+                | Explainee::Index(id)
+                | Explainee::ReplanView(id)
+                | Explainee::ReplanMaterializedView(id)
+                | Explainee::ReplanIndex(id) => btreeset! {*id},
+                Explainee::Statement(stmt) => stmt.depends_on(),
```

Comment:
I think this won't traverse view definitions, because it just looks at HIR and stops at Gets of global ids. This is a problem, because the queries below look for these ids in such introspection relations where views won't be represented, because they are concerned only with objects that have dataflows. E.g., views won't have frontiers or hydration statuses or materialization lags.

You could fix this by gathering the ids in a similar way as timestamp selection: https://github.com/MaterializeInc/materialize/blob/2769566d5c435fd4fa750f7c3f897ab39ab11d2e/src/adapter/src/coord/sequencer/inner/peek.rs#L508-L510

Reply:
Relevant slack thread in the meantime: https://materializeinc.slack.com/archives/C074FURKP8W/p1719615038773579
In that thread, people are saying that it would be good to go all the way to sources, so even recursing through even materialized views, which not even `sufficient_collections` would do. But I guess it's possible to write a similar function to `sufficient_collections` that would go through also materialized views and indexes.